sunpy/io/fits.py

```diff
@@ -200,9 +200,9 @@ def header_to_fits(header):
     key_comments = header.pop('KEYCOMMENTS', False)
 
     for k, v in header.items():
-        # Drop any keys which are too long to save into FITS
-        if len(k) > 8:
-            warnings.warn(f"The meta key {k} is too long, dropping from the FITS header.", SunpyUserWarning)
+        # Drop any keys which are too long to save into FITS, or have non-ascii characters
+        if len(k) > 8 or not fits.Card._ascii_text_re.match(str(v)):
```

Comment:
Would be nice to split these checks and raise a different more descriptive error in each case (instead of "either this or this is wrong").